buzsaki_lab_to_nwb/tingley_code_septal/convert_tingley_septal.py

```diff
@@ -0,0 +1,114 @@
+]
+
+session_path_list = [
+    session
+    for subject in data_path.iterdir()
+    if subject.is_dir() and subject.name in subject_list
+    for session in subject.iterdir()
+    if session.is_dir() and session.name not in invalid_session
+]
+
+# session_path_list = [Path("/shared/catalystneuro/Buzsaki/TingleyD/DT8/20170220_216um_1944um_170220_192456")]
+
+counter = 0
+for session_path in session_path_list:
+    print("----------------")
+    print(session_path)
+    counter += 1
+    print(f"session {counter} of {len(session_path_list)}")
+    session_id = session_path.name
+    lfp_file_path = session_path / f"{session_path.name}.lfp"
+    raw_file_path = session_path / f"{session_id}.dat"
+    xml_file_path = session_path / f"{session_id}.xml"
+    spikes_matfile_path = session_path / f"{session_id}.spikes.cellinfo.mat"
+    session_info_matfile_path = session_path / f"{session_id}.sessionInfo.mat"
+    behavior_matfile_path = session_path / f"{session_id}.behavior.mat"
+    if stub_test:
+        nwbfile_path = nwb_output_path / f"{session_id}_stub.nwb"
+    else:
+        nwbfile_path = nwb_output_path / f"{session_id}.nwb"
+
+    print("raw file available", raw_file_path.is_file())
+    print("lfp file available", lfp_file_path.is_file())
+    print("behavior / position mat file available", behavior_matfile_path.is_file())
+    source_data = dict()
+    conversion_options = dict()
+
+    source_data = dict(
+        NeuroscopeLFP=dict(file_path=str(lfp_file_path), gain=conversion_factor, xml_file_path=str(xml_file_path)),
+    )
+    conversion_options.update(NeuroscopeLFP=dict(stub_test=stub_test))
+
+    if raw_file_path.is_file():
+        source_data.update(
+            NeuroscopeRecording=dict(
+                file_path=str(raw_file_path), gain=conversion_factor, xml_file_path=str(xml_file_path)
+            )
+        )
+        conversion_options.update(NeuroscopeRecording=dict(stub_test=stub_test, es_key="ElectricalSeries_raw"))
+
+    clu_matches_in_session = len(list(session_path.glob("*.clu*")))
+    res_matches_in_session = len(list(session_path.glob("*.res*")))
+    if clu_matches_in_session > 0 and res_matches_in_session > 0:
+        print("neuroscope sorted data available", True)
+        source_data.update(
+            NeuroscopeSorting=dict(
+                folder_path=str(session_path), keep_mua_units=False, xml_file_path=str(xml_file_path)
+            )
+        )
+        conversion_options.update(NeuroscopeSorting=dict(stub_test=stub_test))
+
+    if spikes_matfile_path.is_file():
+        try:
+            print("spikes and sessionInfo matlab files available", spikes_matfile_path.is_file())
+            loadmat(spikes_matfile_path)
+            loadmat(session_info_matfile_path)
+            # source_data.update(CellExplorerSorting=dict(spikes_matfile_path=str(spikes_matfile_path)))
+            source_data.update(CellExplorerSorting=dict(file_path=str(spikes_matfile_path)))
```

Comment:
@h-mayorquin Can you nest this logic in the following way?

-> If CellExplorer data is present, add that to `source_data`
-> else, if Neuroscope data, add that to `source_data`

Under the broken `nwb-conversion-tools==0.10.3`, this was actually causing the `Units` columns to be corrupted with `Nan` values because the Neuroscope was overwriting in an append-like fashion.

Now that we're back to `nwb-conversion-tools==0.9.5`, it has a slightly safer behavior, but the Neuroscope is apparently overwriting the table completely, removing the custom column properties we want from CellExplorer.

Reply:
Makes sense, done.